oximeter/oximeter/schema/database-transaction.toml

```diff
@@ -0,0 +1,26 @@
+format_version = 1
+
+[target]
+name = "database_transaction"
+description = "A named transaction run in the control plane database"
+authz_scope = "fleet"
+versions = [
+    { version = 1, fields = [ "name" ] }
+]
+
+[[metrics]]
+name = "retry_data"
+description = "Information about a retried transaction"
+units = "bytes"
+datum_type = "f64"
```

Comment:
`RetryData` previously included a `latency` datum (f64) which was "seconds".

I'm not sure I'm seeing where that's represented here. I do see the `attempt` number below on line 25ish, but is it  right to say that the units are "byte" here?

Reply:
Ah, the unit is wrong, thanks for catching that. The `latency` field is now `datum`. I did not add the ability to rename the actual datum field in the TOML definitions, which is what we'd done here with the `#[datum]` attribute on the `latency` field.

Reply:
I'm going to queue this behind #6006, which adds the `Unit::Seconds` variant that this needs. I'll rebase and merge after that integrates into `main`.

Reply:
Picked up the seconds variant in `1b77125f5`. I also found and fixed a bug in how extra traits are derived for the generated types. That is enough code that it deserves a second look @smklein. Thanks!